### PR TITLE
Update docs to include drawPoints API.md

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Ran `yarn docs`
  - Updated API.md to include `drawPoints`

#### Checklist

- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -14,20 +16,38 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
+*   `data` **([Array][12]\<Coordinate> | [Array][12]\<Feature>)** An array of coordinate data or GeoJSON Features used as the data source for maplibre
+*   `map` **maplibreMap** A maplibre map on which the points will be drawn
+*   `options` **[Object][13]** An object containing options for changing the styles and features of the points rendered to the map, see the options for more details on available settings
+
+    *   `options.showCluster` **[String][7]** Determines whether or not points close together should be clustered into a single point (optional, default `true`)
+    *   `options.clusterOptions` **[String][7]** Object for determining cluster options, see [ClusterOptions][14] for more details (optional, default `{}`)
+    *   `options.unclusteredOptions` **[String][7]** Object for determining cluster options, see [UnclusteredOptions][15] for more details (optional, default `{}`)
+*   `mapStyle` **[String][7]** A string containing the map style data used to draw this map
+
+Returns **DrawPointsOutput** An object the string id's of the sources and layers used to draw the points to the map
```
Comment (Amplifiyer) [inline API.md:50]: We should expand this and talk about all the Ids created or returned.

File: API.md [hunk 2]
```diff
@@ -14,20 +16,38 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
+### Parameters
+
+*   `sourceName` **[String][7]** A user defined name used for determining the maplibre data source and the maplibre layers
+*   `data` **([Array][12]\<Coordinate> | [Array][12]\<Feature>)** An array of coordinate data or GeoJSON Features used as the data source for maplibre
+*   `map` **maplibreMap** A maplibre map on which the points will be drawn
+*   `options` **[Object][13]** An object containing options for changing the styles and features of the points rendered to the map, see the options for more details on available settings
+
+    *   `options.showCluster` **[String][7]** Determines whether or not points close together should be clustered into a single point (optional, default `true`)
+    *   `options.clusterOptions` **[String][7]** Object for determining cluster options, see [ClusterOptions][14] for more details (optional, default `{}`)
+    *   `options.unclusteredOptions` **[String][7]** Object for determining cluster options, see [UnclusteredOptions][15] for more details (optional, default `{}`)
```
Comment (Amplifiyer) [inline API.md:47]: Is this accurate "Object for determining cluster options" here?

File: API.md [hunk 3]
```diff
@@ -14,20 +16,38 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
+
+## drawPoints
+
+DrawPoints utility function for adding points to a map based on coordinate data or a FeatureCollection. Will add clustered points and styled markers by default with options for popups and other styles
+
+### Parameters
+
+*   `sourceName` **[String][7]** A user defined name used for determining the maplibre data source and the maplibre layers
+*   `data` **([Array][12]\<Coordinate> | [Array][12]\<Feature>)** An array of coordinate data or GeoJSON Features used as the data source for maplibre
+*   `map` **maplibreMap** A maplibre map on which the points will be drawn
```
Comment (Amplifiyer) [inline API.md:42]:
```suggestion
*   `map` **maplibre-gl-js Map** A maplibre map on which the points will be drawn
```

File: API.md [hunk 4]
```diff
@@ -14,20 +16,38 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
+*   `options` **[Object][13]** An object containing options for changing the styles and features of the points rendered to the map, see the options for more details on available settings
+
+    *   `options.showCluster` **[String][7]** Determines whether or not points close together should be clustered into a single point (optional, default `true`)
+    *   `options.clusterOptions` **[String][7]** Object for determining cluster options, see [ClusterOptions][14] for more details (optional, default `{}`)
+    *   `options.unclusteredOptions` **[String][7]** Object for determining cluster options, see [UnclusteredOptions][15] for more details (optional, default `{}`)
+*   `mapStyle` **[String][7]** A string containing the map style data used to draw this map
```
Comment (Amplifiyer) [inline API.md:48]: How is mapStyle used in the `drawPoints`?

Comment (thaddmt) [reply API.md:48]: It is used for determining the default font set to use, I can add that to the comments.

File: API.md [hunk 5]
```diff
@@ -16,38 +17,45 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
-    *   `options.clusterOptions` **[String][7]** Object for determining cluster options, see [ClusterOptions][14] for more details (optional, default `{}`)
-    *   `options.unclusteredOptions` **[String][7]** Object for determining cluster options, see [UnclusteredOptions][15] for more details (optional, default `{}`)
-*   `mapStyle` **[String][7]** A string containing the map style data used to draw this map
+    *   `options.showCluster` **[String][8]** Determines whether or not points close together should be clustered into a single point (optional, default `true`)
+    *   `options.clusterOptions` **[String][8]** Object for determining cluster options, see [ClusterOptions][15] for more details (optional, default `{}`)
+    *   `options.unclusteredOptions` **[String][8]** Object for determining unclustered point options, see [UnclusteredOptions][16] for more details (optional, default `{}`)
+*   `mapStyle` **[String][8]** The map style returned from Amplify Geo that is currently being used. This is used to determine the default fonts used by maplibre-gl-js.
```
Comment (Amplifiyer) [inline API.md:49]: It's still not clear to me from the description. Is this a required param?

Can we not retrieve it from the `maplibre-gl-js-map` instance?

File: API.md [hunk 6]
```diff
@@ -16,38 +17,45 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
 
-Returns **DrawPointsOutput** An object the string id's of the sources and layers used to draw the points to the map
+### Properties
+
+*   `sourceId` **[String][8]** 
+*   `clusterLayerId` **[String][8]** 
+*   `clusterSymbolLayerId` **[String][8]** 
+*   `unclusteredLayerId` **[String][8]** 
+
+Returns **DrawPointsOutput** output An object containing the string id's of the sources and layers used to draw the points to the map. This includes the sourceId, clusterLayerId, clusterSymbolLayerId, unclusteredLayerId.
```
Comment (Amplifiyer) [inline API.md:58]: Should the properties be after the `DrawPointsOutput` definition? And possible a definition/comment for each of these Ids.

File: API.md (hunk 5 above)
Comment (thaddmt) [reply API.md:49]: For now this is a required parameter when using `drawPoints` with Amazon location services maps.

The map style is referring to a string like `VectorEsriNavigation` which is not returned by maplibre's [getStyle](https://maplibre.org/maplibre-gl-js-docs/api/map/#map#getstyle)

File: API.md (hunk 6 above)
Comment (thaddmt) [reply API.md:58]: Because of the way the documentation js works I can't get those properties to come after the `Return` section

File: API.md [hunk 7]
```diff
@@ -14,34 +17,88 @@ An object for encapsulating an Amplify Geo transform request and Amplify credent
...
+
+[18]: https://maplibre.org/maplibre-gl-js-docs/api/sources/
+
+[19]: https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/
+
+[20]: https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#symbol
+
+[21]: https://maplibre.org/maplibre-gl-js-docs/style-spec/layers
+
+[22]: https://github.com/maplibre/maplibre-gl-geocoder
```
Comment (TreTuna) [inline API.md:104]: Why did this add all the extra returns, but not for the first 4 items?

**Review by TreTuna — APPROVED** [review]
Just one non-blocking comment here.

**Pull request merged**